src/MagnumPlugins/CgltfImporter/CgltfImporter.cpp

```diff
@@ -2523,6 +2556,463 @@ Containers::Optional<MaterialData> CgltfImporter::doMaterial(const UnsignedInt i
+                if(textureView.scale != 1.0f) {
+                    Utility::formatInto(nameBuffer, "{}Scale", name);
+                    const Containers::StringView scaleName = nameBuffer.prefix(name.size() + 5);
+                    if(checkMaterialAttributeSize(scaleName, MaterialAttributeType::Float))
+                        arrayAppend(extensionAttributes, InPlaceInit,
+                            scaleName, textureView.scale);
+                }
+
+                /* We added all attributes, nothing left to do */
+                continue;
+
+            /* A primitive is anything that's not a string, object or array. We
+               ignore non-primitive arrays, so we can handle both in one place. */
+            } else if(token.type == JSMN_PRIMITIVE || token.type == JSMN_ARRAY) {
+                const UnsignedInt start = tokenIndex + UnsignedInt(token.type == JSMN_ARRAY);
+                /* Primitive token size is 0, but we can't use max() or we'd
+                   allow empty arrays */
+                const UnsignedInt count = token.type == JSMN_PRIMITIVE ? 1 : token.size;
+
+                /* Using a lambda here to make aborting early easier */
+                auto attributeType = [&]() -> MaterialAttributeType {
+                    /* No use importing arbitrarily-sized arrays of primitives,
+                       those are currently not used in any glTF extension */
+                    if(count < 1 || count > 4) return MaterialAttributeType{};
+
+                    MaterialAttributeType finalType{};
+                    /* This still works for non-primitive array members (like
+                       objects or nested arrays), because we instantly abort
+                       when we find one of those, so we never need more than
+                       four tokens */
+                    for(const jsmntok_t& element: tokens.slice(start, start + count)) {
+                        if(element.type != JSMN_PRIMITIVE)
+                            return MaterialAttributeType{};
+
+                        const Containers::StringView value = tokenString(json, element);
+
+                        /* Jsmn only checks the first character, which allows
+                           some invalid values like nnn. We perform some basic
+                           type detection and invalid values result in 0. This
+                           matches cgltf behaviour.
+                           Technically integers can use exponent notation, too:
+                           https://www.khronos.org/registry/glTF/specs/2.0/glTF-2.0.html#json-encoding
+                           But cgltf_json_to_int() doesn't parse them correctly
+                           and they should be incredibly rare in the wild, so
+                           we always interpret them as float. */
+                        /** @todo Override CGLTF_ATOI with a parsing function
+                            that handles integers with exponent notation */
```

Comment:
The nice part is that this doesn't break texture attributes, since we parse texture objects differently. We already know to expect integers for texture indices and coordinate sets. The cgltf-handled extensions don't have any integers, so nothing awkward happening there, either. I just added a comment to make sure this is handled for any possible future cases.